tests/test_entrypoints.py

```diff
@@ -0,0 +1,265 @@
+container_uuid = "123e4567-e89b-12d3-a456-426614174001"
+
+
+def test_dlhub_entrypoint(
+    mocker,
+    faker,
+):
+    mock_executor = mocker.patch("garden_ai.entrypoints.globus_compute_sdk.Executor")
+    dlhub_entrypoint_metadata = RegisteredEntrypointMetadata(
+        doi="10.26311/3hz8-as26",  # A DOI in the DLHub list
+        title="Migrated DLHub Model",
+        short_name="run_dlhub_model",
+        authors=["Joe Schmoe"],
+        func_uuid=function_uuid,
+        container_uuid=function_uuid,
+        base_image_uri=faker.url(),
+        full_image_uri=faker.url(),
+        notebook_url=faker.url(),
+        function_text=faker.text(),
+    )
+
+    dlhub_entrypoint = Entrypoint(metadata=dlhub_entrypoint_metadata)
+
+    dlhub_wrapped_result = (("answer", {"stdout": None, "success": True}), 3452354)
+
+    mock_executor_instance = mocker.MagicMock()
+    mock_executor.return_value.__enter__.return_value = mock_executor_instance
+    mock_executor_instance.submit_to_registered_function.return_value.result.return_value = (
+        dlhub_wrapped_result
+    )
+
+    result = dlhub_entrypoint("test arg", endpoint="specific-endpoint")
+
+    assert (
+        result == "answer"
+    ), "Should return the direct result for an entrypoint with DOI in list"
+    mock_executor_instance.submit_to_registered_function.assert_called_once()
+    mock_executor_instance.submit_to_registered_function.assert_called_with(
+        args=(
+            {
+                "inputs": "test arg",
+                "parameters": [],
+                "debug": False,
+            },
+        ),
+        function_id=function_uuid,
+        kwargs={},
+    )
+
+
+def test_normal_entrypoint(
+    mocker,
+    faker,
+):
+    mock_executor = mocker.patch("garden_ai.entrypoints.globus_compute_sdk.Executor")
+    normal_entrypoint_metadata = RegisteredEntrypointMetadata(
+        doi="foo",  # A DOI not in the DLHub list
+        title="Some cool model",
+        short_name="run_model",
+        authors=["Jane Schmane"],
+        func_uuid=function_uuid,
+        container_uuid=function_uuid,
+        base_image_uri=faker.url(),
+        full_image_uri=faker.url(),
+        notebook_url=faker.url(),
+        function_text=faker.text(),
+    )
+
+    normal_entrypoint = Entrypoint(metadata=normal_entrypoint_metadata)
+
+    mock_executor_instance = mocker.MagicMock()
+    mock_executor.return_value.__enter__.return_value = mock_executor_instance
+    mock_executor_instance.submit_to_registered_function.return_value.result.return_value = (
+        "mocked result"
+    )
+
+    result = normal_entrypoint("test arg", endpoint="specific-endpoint")
+
+    assert (
+        result == "mocked result"
+    ), "Should return the direct result for an entrypoint with DOI not in list"
+    mock_executor_instance.submit_to_registered_function.assert_called_once()
+
+
+def test_entrypoint_test_raises_on_non_idempotent_entrypoint(mocker):
+    class NonIdempotentCounter:
```

Comment:
just an understanding check bc I noticed this one is a bit longer than the others -- was there something especially tricky to mock/test here that we may want to revisit design-wise? or is this one just long because it's thorough, and wouldn't have benefit from splitting into smaller tests